File: 07-langs_particularities/2-java/src/main/java/tv/codely/app/api/PostLikePutController.java
package tv.codely.app.api;

import tv.codely.contexts.rrss.post_likes.application.like.PostLiker;
import tv.codely.contexts.rrss.posts.domain.PostDoesNotExistError;
import tv.codely.contexts.rrss.users.domain.UserDoesNotExistError;
import tv.codely.contexts.shared.infrastructure.Response;

public record PostLikePutController(PostLiker postLiker) {
    public Response on(String id, String postId, String likerUserId) {
        try {
            postLiker.like(id, postId, likerUserId);

            return new Response(201, "");
        } catch (PostDoesNotExistError | UserDoesNotExistError error) {
            return new Response(409, error.getMessage());
        }
    }
}

File: 07-langs_particularities/1-php/src/App/Api/PostLikePutController.php
<?php
declare(strict_types=1);

namespace CodelyTv\App\Api;

use CodelyTv\Contexts\PostLikes\Application\Like\PostLiker;
use CodelyTv\Contexts\Posts\Domain\PostDoesNotExistError;
use CodelyTv\Contexts\Shared\Infrastructure\Response;
use CodelyTv\Contexts\Users\Domain\UserDoesNotExistError;

final readonly class PostLikePutController
{
	public function __construct(private PostLiker $postLiker)
	{
	}

	public function __invoke(string $id, string $postId, string $likerUserId): Response
	{
		try {
			$this->postLiker->like($id, $postId, $likerUserId);

			return new Response(201);
		} catch (PostDoesNotExistError|UserDoesNotExistError $error) {
			return new Response(409, $error->getMessage());
		}
	}
}
